pallets/creditcoin/src/helpers.rs

```diff
@@ -113,4 +114,69 @@ impl<T: Config> Pallet<T> {
 		UsedGuids::<T>::insert(guid, ());
 		Ok(())
 	}
+
+	pub fn register_transfer_internal(
+		who: T::AccountId,
+		from_id: AddressId<T::Hash>,
+		to_id: AddressId<T::Hash>,
+		transfer_kind: TransferKind,
+		amount: ExternalAmount,
+		order_id: OrderId<T::BlockNumber, T::Hash>,
+		blockchain_tx_id: ExternalTxId,
+	) -> Result<
+		(TransferId<T::Hash>, Transfer<T::AccountId, BlockNumberFor<T>, T::Hash>),
+		crate::Error<T>,
+	> {
+		let from = Self::get_address(&from_id)?;
+		let to = Self::get_address(&to_id)?;
+
+		ensure!(from.owner == who, Error::<T>::NotAddressOwner);
+
+		ensure!(from.blockchain == to.blockchain, Error::<T>::AddressPlatformMismatch);
+
+		ensure!(from.blockchain.supports(&transfer_kind), Error::<T>::UnsupportedTransferKind);
+
+		let transfer_id = TransferId::new::<T>(&from.blockchain, &blockchain_tx_id);
+		ensure!(!Transfers::<T>::contains_key(&transfer_id), Error::<T>::TransferAlreadyRegistered);
+
+		if *blockchain_tx_id == *b"0" {
+			// this transfer is an exemption, no need to verify it
+			let transfer = Transfer {
+				blockchain: from.blockchain,
+				kind: transfer_kind,
+				amount: ExternalAmount::zero(),
```

Comment:
This is not specific to this PR, is it? This should be applied regardless if this PR passes or not. Was it a bug? My understanding is there was some use in exempting with a specified amount but, it makes little sense if the transfer does not really exist.

Reply:
Yeah it was a bug, but I'm going to remove this logic soon anyways.

We never really supported exempting a specified amount, because you can only have one "repayment" transfer on a deal order, and exemption requires filling that transfer with the dummy - so you couldn't have a partial repayment. We'd need to support repayment across multiple transfers for that.